File: 실버 3 (5문제)/2579_계단 오르기.py
# 참고 링크 : https://moon-gd.tistory.com/55
import sys
q_input = sys.stdin.readline  # 빠른 입력을 위해서

size = int(q_input())

scores = [0] * (size + 1)
stairs = [0] * (size + 1)

# 입력 받기
for i in range(1, size + 1):
    scores[i] = int(q_input())

# size == 1 인 경우, Easy
stairs[1] = scores[1]

# size > 1 인 경우 DP를 통해서 계산
if size > 1:
    stairs[2] = scores[1] + scores[2]

    for i in range(3, size + 1):
        stairs[i] = max(stairs[i - 2] + scores[i], stairs[i - 3] + scores[i - 1]+scores[i])

print(stairs[size])

File: 실버 3 (6문제)/2579_계단 오르기.c
// 참고 링크 : https://moon-gd.tistory.com/55
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
#define MAX 301
#define get_max(a, b) a > b ? a : b

int scores[MAX];
int stairs[MAX];

int main() {
	int size;
	scanf("%d", &size);

	// 점수 입력 받기
	for (int i = 1; i <= size; i++) {
		scanf("%d", &scores[i]);
	}

	stairs[1] = scores[1];
	stairs[2] = scores[1] + scores[2];

	for (int i = 3; i <= size; i++) {
		stairs[i] = get_max(stairs[i - 2] + scores[i], stairs[i - 3] + scores[i - 1] + scores[i]);
	}

	printf("%d", stairs[size]);

	return 0;
}